modules/videoio/src/cap_libcamera.cpp

```diff
@@ -0,0 +1,1100 @@
+        break;
+    }
+
+    case cv::CAP_PROP_FRAME_WIDTH:
+        options->video_width = options->photo_width = (int)value;
+        needsReconfigure.store(true, std::memory_order_release);
+        break;
+
+    case cv::CAP_PROP_FRAME_HEIGHT:
+        options->video_height = options->photo_height = (int)value;
+        needsReconfigure.store(true, std::memory_order_release);
+        break;
+
+    case cv::CAP_PROP_FPS:
+        options->framerate = (float)value;
+        needsReconfigure.store(true, std::memory_order_release);
+        break;
+    case cv::CAP_PROP_AUTOFOCUS:  // Not implemented
+    case cv::CAP_PROP_BUFFERSIZE: // Not implemented
+    case cv::CAP_PROP_PAN:        // Not implemented
+    case cv::CAP_PROP_TILT:       // Not implemented
+    case cv::CAP_PROP_ROLL:       // Not implemen ted
+    case cv::CAP_PROP_IRIS:       // Not implemented
+        // These properties might need to trigger a re-configuration of the camera.
+        // You can handle them here if you want to support changing resolution or framerate on-the-fly.
+        // For now, we'll return false to indicate that these properties are not supported for dynamic changes.
+        std::cerr << "Warning: Property " << propId << " is not supported for dynamic changes." << std::endl;
+        return false;
+
+    default:
+        std::cerr << "Warning: Unsupported property: " << propId << std::endl;
+        return false;
+    }
+
+    // if (needsReconfigure)
+    // {
+    //     if (isFramePending)
+    //     {
+    //         stopVideo();
+    //         startVideo();
+    //     }
+    // }
+    return true;
+}
+
+bool LibcameraCapture::open(int _index)
+{
+    cv::String name;
+    /* Select camera, or rather, V4L video source */
+    if (_index < 0) // Asking for the first device available
+    {
+        for (int autoindex = 0; autoindex < 8; ++autoindex) // 8=MAX_CAMERAS
+        {
+            name = cv::format("/dev/video%d", autoindex);
+            /* Test using an open to see if this new device name really does exists. */
+            int h = ::open(name.c_str(), O_RDONLY);
+            if (h != -1)
+            {
+                ::close(h);
+                _index = autoindex;
+                break;
+            }
+        }
+        if (_index < 0)
+        {
+            CV_LOG_WARNING(NULL, "VIDEOIO(Libcamera): can't find camera device");
+            name.clear();
+            return false;
+        }
+    }
+    else
+    {
+        name = cv::format("/dev/video%d", _index);
+    }
+
+    bool res = open(name);
+    if (!res)
+    {
+        CV_LOG_WARNING(NULL, "VIDEOIO(Libcamera:" << name << "): can't open camera by index");
+    }
+    return res;
+}
+
+bool LibcameraCapture::open(const std::string &_deviceName)
+{
+    CV_LOG_DEBUG(NULL, "VIDEOIO(Libcamera:" << _deviceName << "): opening...");
+    // Some parameters initialization here, maybe more needed.
+    options->video_width = 1280;
+    options->video_height = 720;
```

Comment:
V4L2 uses resolution 640x480 by default as most common. Not all cameras support 720p.